experiments/dice_experiment/run_dice_experiment.py

```diff
@@ -0,0 +1,553 @@
+    index_df, dice_paths_df = experiment_utils.format_results(
+        run_config, dice_paths_df
+    )
+    return {
+        "experiment_config_df": index_df,
+        "dice_paths_df": dice_paths_df,
+    }
+
+
+def merge_results(
+    all_results: Optional[Mapping[str, pd.DataFrame]],
+    run_results: Mapping[str, pd.DataFrame],
+):
+    """Concatenates newly generated result dataframes with previously generated
+    result dataframes.
+
+    Args:
+        all_results: The previously generated results.
+        run_results: The newly generated results to merge into the previous
+            results.
+
+    Returns:
+        A merged set of results containing data from all_results and
+        run_results."""
+    if not all_results:
+        return run_results
+    else:
+        if set(all_results.keys()) != set(run_results.keys()):
+            raise RuntimeError(
+                (
+                    "The new results dictionary contains keys "
+                    f"{run_results.keys()}, but the original results have keys"
+                    f" {all_results.keys()}"
+                )
+            )
+
+        merged_results: Mapping[str, pd.DataFrame] = {}
+        for result_name in run_results.keys():
+            original_result = all_results[result_name]
+            new_result = run_results[result_name]
+            merged_results[result_name] = pd.concat(
+                [original_result, new_result]
+            ).reset_index(drop=True)
+
+        return merged_results
+
+
+# TODO(@jakeval): Unit test this eventually.
+def save_results(
+    results: Mapping[str, pd.DataFrame],
+    results_directory: Optional[str],
+    config: Mapping[str, Any],
+    only_csv: bool = False,
+) -> str:
+    """Saves the results and experiment config to the local file system.
+
+    Args:
+        results: A mapping from filename to DataFrame.
+        results_directory: Where to save the results.
+        config: The config used to run this experiment.
+        only_csv: Whether to exclude saving the config.json file.
+
+    Returns:
+        The directory where the results are saved."""
+    if not results_directory:
+        results_directory = os.path.join(
+            _RESULTS_DIR, config["experiment_name"]
+        )
+    if os.path.exists(results_directory):
+        shutil.rmtree(results_directory)
+    os.makedirs(results_directory)
+    for result_name, result_df in results.items():
+        result_df.to_csv(
+            os.path.join(results_directory, result_name + ".csv"), index=False
+        )
+    if not only_csv:
+        with open(
+            os.path.join(results_directory, "config.json"), "w"
+        ) as config_file:
+            json.dump(config, config_file)
+    return results_directory
+
+
+def _get_results_dir(results_directory, experiment_name):
+    return results_directory or os.path.join(_RESULTS_DIR, experiment_name)
+
+
+def run_batch(
+    run_configs: Sequence[Mapping[str, Any]],
```

Comment:
Aligned with the other experimental code for MRMC, we may want to consider renaming `run` to something more explicit. In this context, a `run` is a single execution of the MRMC pipeline, correct? Do you think something like `single_mrmc_config` is a clearer terminology?

Note that this renaming isn't a blocker to submit this PR, though if we don't already have an Issue or clearly marked TODO for this you should add one.

Reply:
I've added a TODO linking to [Issue 41](https://github.com/jakeval/MRMC/issues/41).